righttyper/typeinfo.py

```diff
@@ -204,7 +220,9 @@ def rebuild(types: tuple[TypeInfo, ...]) -> TypeInfo:
                 for i in range(len(types[0].args))
             )
 
-            return types[0].replace(args=args)
+            homogenized = types[0].replace(args=args)
```

Comment:
Do we really need these? Won't `merged_types` below do it anyway?
Is there a test that fails if you keep the old code from line 207?